src/Runner.Common/Util/NodeUtil.cs

```diff
@@ -15,8 +15,21 @@ public static class NodeUtil
 
         public static string GetInternalNodeVersion()
         {
-            var forcedNodeVersion = Environment.GetEnvironmentVariable(Constants.Variables.Agent.ForcedInternalNodeVersion);
-            return !string.IsNullOrEmpty(forcedNodeVersion) && BuiltInNodeVersions.Contains(forcedNodeVersion) ? forcedNodeVersion : _defaultNodeVersion;
+            var forcedInternalNodeVersion = Environment.GetEnvironmentVariable(Constants.Variables.Agent.ForcedInternalNodeVersion);
+            var forcedNodeVersion = Environment.GetEnvironmentVariable(Constants.Variables.Agent.ForcedActionsNodeVersion);
```

Comment:
This method is for determining the node version for **internal** JS processes, like hashfiles.js or hook execution. This has no effect on how a nodeJS github action is executed.

Why do you consider the forced **action** node version here?

Reply:
I think that you're right, now it makes more sense. I assume I misunderstood Thomas' request, thank you for your explanation.